convergence/do_all_the_convergence_tests/error_table.py

```diff
@@ -0,0 +1,45 @@
+from argparse import ArgumentParser
+import numpy as np
+import pandas as pd
+
+
+def calculate_error_rates(errors_df):
+    rate_dicts = []
+    for n in pd.unique(errors_df["norm"]):
+        for v in pd.unique(errors_df["var"]):
+            conv_df = errors_df[(errors_df["norm"] == n) & (errors_df["var"] == v)]
+            d = {"norm": n, "var": v}
+            resolutions = pd.unique(conv_df["h"])
+
+            for r in range(len(resolutions) - 1):
+                error_decay = (
+                    conv_df.loc[conv_df["h"] == resolutions[r], "error"].values[0]
+                    / conv_df.loc[conv_df["h"] == resolutions[r + 1], "error"].values[0]
+                )
+                rate = np.log(error_decay) / np.log(resolutions[r] / resolutions[r + 1])
+                resolution_decay = "{}->{}".format(
+                    np.round(resolutions[r], 3), np.round(resolutions[r + 1], 3)
+                )
+                d.update({resolution_decay: rate})
+            rate_dicts.append(d)
+
+    rate_df = pd.DataFrame(rate_dicts)
+    return rate_df
```

Comment:
You can compute convergence in this way, a more reliable way is to compute the convergence via linear regression (log(h) vs log(error))

Reply:
What do you mean with `more reliable`? Take for example a convergence test in single precision, which will come to a plateau pretty soon. Then you can see the correct rates in the beginning and the plateau later. With the regression, you only see the overall result.
Nonetheless I've added the regression feature ;-)